kvbc/include/categorization/details.h

```diff
@@ -0,0 +1,24 @@
+#include <vector>
+#include <string>
+#include "categorized_kvbc_msgs.cmf.hpp"
+#include "kv_types.hpp"
+#include "sparse_merkle/base_types.h"
+
+namespace concord::kvbc::categorization {
+
+namespace serialization {
+using buffer = std::vector<uint8_t>;
+}  // namespace serialization
+
+// DB key is a CMF stuct of hashed key and the version.
+// On serialization it will be hashed_key || version(big endian)
+DbKey genDbKey(const std::string& key, const BlockId& version) {
```

Comment:
Is that function related to the merkle category? If not, maybe we can be explicit about which hash it uses (i.e. `concord::util::SHA3_256`. Also, maybe we can define a hash type to reuse across `categorization`.

Reply:
it's not related to the Merkle, it's related to all but the Merkle :) as they can use the new CMF.
So yes we can define a category hash type and use it.